### PR TITLE
Consolidate full files downloads

### PR DESCRIPTION
**Pull request by otaviobp**
- Consolidate full files downloads from verify and update.
 - Use the consolidated function on bundle-add
 - I'm still working on verify --install

Related to #447

### CODE REVIEW

File: src/verify.c [hunk 1]
```diff
@@ -348,125 +348,128 @@ static int get_all_files(struct manifest *official_manifest, struct list *subs)
...
-				continue;
 			}
-			full_download(file);
-		} else {
-			/* mark the file as good to save time later */
-			file->do_not_update = 1;
+			untar_full_download(file);
+			free_string(&filename);
+			free_string(&url);
+			continue;
```
Comment (matthewrsj) [inline src/verify.c]: Hmm, not your fault, but I notice we aren't properly printing progress for our mixed files.

Comment (otaviobp) [reply src/verify.c]: Good point. I'll add that in a future commit.

File: src/verify.c [hunk 2]
```diff
@@ -348,125 +348,128 @@ static int get_all_files(struct manifest *official_manifest, struct list *subs)
...
+	iter = list_head(files);
+	while (iter) {
+		struct file *f = iter->data;
+		char *fullname;
+		bool valid;
+
+		print_progress(complete, total);
+		complete++;
+		iter = iter->next;
+		if (f->is_deleted || f->do_not_update)
```
Comment (matthewrsj) [inline src/verify.c]: I prefer to see brackets even on one-line ifs. I notice you do use brackets elsewhere and the rest of the codebase should be as well.

File: src/verify.c [hunk 3]
```diff
@@ -348,104 +348,19 @@ static int get_all_files(struct manifest *official_manifest, struct list *subs)
...
-error:
-	fprintf(stderr, "Error: Unable to download neccessary files for this OS release\n");
-	return -EFULLDOWNLOAD;
-}
-
 /*
  * Check if the hash of all files in the list matches the system and in this case mark
  * them as do_not_update.
  * If cmdline_option_quick is set, use lazy hash mode and just check if file exists.
+ * Return 0 if any file in the list of files doesn't match the system. Returns a
```
Comment (matthewrsj) [inline src/verify.c]: This seems backwards to me but ok.

File: src/verify.c [hunk 4]
```diff
@@ -482,8 +401,13 @@ static int get_required_files(struct manifest *official_manifest, struct list *s
 	if (cmdline_option_fix) {
 		int ret;
 
-		check_files_hash(official_manifest->files);
-		ret = get_missing_files(official_manifest->files);
+		if (check_files_hash(official_manifest->files))
```
Comment (matthewrsj) [inline src/verify.c]: brackets please

File: test/functional/verify/boot-file-deleted/lines-checked [hunk 5]
```diff
@@ -1,7 +1,5 @@
 Verifying version 10
 Verifying files
-Starting download of remaining update content. This may take a while...
```
Comment (matthewrsj) [inline test/functional/verify/boot-file-deleted/lines-checked]: nice to see these go away

**Comment by otaviobp** [comment]
Github messes up the git history. 56ecaf0 comes after 277781c

File: src/verify.c (hunk 2 above)
Comment (otaviobp) [reply src/verify.c]: my bad

File: src/verify.c [hunk 6]
```diff
@@ -394,25 +394,21 @@ static int check_files_hash(struct list *files)
...
+	int ret;
+
 	if (cmdline_option_install) {
-		return get_all_files(official_manifest, subs);
+		get_all_files(official_manifest, subs);
 	}
 
-	if (cmdline_option_fix) {
-		int ret;
+	if (check_files_hash(official_manifest->files))
```
Comment (matthewrsj) [inline src/verify.c]: brackets please

**Comment by otaviobp** [comment]
Branch updated with fixes:
 - Improving comment of function check_files_hash()
 - Fixing style issues reported by make compliant
 - Fixing missing brackets

**Pull request merged**